test/packages/rust/src/generated/instructions/approve_collection_authority.rs

```diff
@@ -0,0 +1,383 @@
+    update_authority: Option<solana_program::pubkey::Pubkey>,
+    payer: Option<solana_program::pubkey::Pubkey>,
+    metadata: Option<solana_program::pubkey::Pubkey>,
+    mint: Option<solana_program::pubkey::Pubkey>,
+    system_program: Option<solana_program::pubkey::Pubkey>,
+    rent: Option<solana_program::pubkey::Pubkey>,
+}
+
+impl ApproveCollectionAuthorityBuilder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+    pub fn collection_authority_record(
+        &mut self,
+        collection_authority_record: solana_program::pubkey::Pubkey,
+    ) -> &mut Self {
+        self.collection_authority_record = Some(collection_authority_record);
+        self
+    }
+    pub fn new_collection_authority(
+        &mut self,
+        new_collection_authority: solana_program::pubkey::Pubkey,
+    ) -> &mut Self {
+        self.new_collection_authority = Some(new_collection_authority);
+        self
+    }
+    pub fn update_authority(
+        &mut self,
+        update_authority: solana_program::pubkey::Pubkey,
+    ) -> &mut Self {
+        self.update_authority = Some(update_authority);
+        self
+    }
+    pub fn payer(&mut self, payer: solana_program::pubkey::Pubkey) -> &mut Self {
+        self.payer = Some(payer);
+        self
+    }
+    pub fn metadata(&mut self, metadata: solana_program::pubkey::Pubkey) -> &mut Self {
+        self.metadata = Some(metadata);
+        self
+    }
+    pub fn mint(&mut self, mint: solana_program::pubkey::Pubkey) -> &mut Self {
+        self.mint = Some(mint);
+        self
+    }
+    pub fn system_program(&mut self, system_program: solana_program::pubkey::Pubkey) -> &mut Self {
+        self.system_program = Some(system_program);
+        self
+    }
+    pub fn rent(&mut self, rent: solana_program::pubkey::Pubkey) -> &mut Self {
+        self.rent = Some(rent);
+        self
+    }
+    pub fn build(&self) -> solana_program::instruction::Instruction {
+        let accounts = ApproveCollectionAuthority {
+            collection_authority_record: self
+                .collection_authority_record
+                .expect("collection_authority_record is not set"),
+
+            new_collection_authority: self
+                .new_collection_authority
+                .expect("new_collection_authority is not set"),
+
+            update_authority: self.update_authority.expect("update_authority is not set"),
+
+            payer: self.payer.expect("payer is not set"),
+
+            metadata: self.metadata.expect("metadata is not set"),
+
+            mint: self.mint.expect("mint is not set"),
+
+            system_program: self.system_program.expect("system_program is not set"),
+
+            rent: self.rent,
+        };
+        accounts.instruction()
+    }
+}
+
+pub mod cpi {
+    use super::*;
+
+    /// `approve_collection_authority` CPI instruction.
+    pub struct ApproveCollectionAuthority<'a> {
+        pub program: &'a solana_program::account_info::AccountInfo<'a>,
```

Comment:
```suggestion
    pub struct ApproveCollectionAuthority<'a, 'info> {
        pub program: &'a solana_program::account_info::AccountInfo<'info>,
```

Making the `AccountInfo` reference lifetime as long as the `AccountInfo`'s data buffer lifetime results in a lifetime error:

```
error: lifetime may not live long enough
 --> onchain-consumer/src/lib.rs:8:23
  |
7 | pub fn etp(program_id: &Pubkey, accounts: &[AccountInfo], instruction_data: &[u8]) -> ProgramResult {
  |                                 --------  - let's call the lifetime of this reference `'1`
  |                                 |
  |                                 has type `&[AccountInfo<'2>]`
8 |     let mut builder = AddConfigLinesBuilder::new(&accounts[0]);
  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ argument requires that `'1` must outlive `'2`
```